app/controllers/events_controller.rb

```diff
@@ -2,9 +2,19 @@ class EventsController < ApplicationController
 
   def index
     @events = Event.all
+    if params[:topics] || params[:times] || params[:days] || params[:frequencies]
+      @events = Event.tagged_with(params[:topics]||params[:times] || params[:days] ||params[:frequencies], :any => true)
+    end
+    
+    @topics = ActsAsTaggableOn::Tagging.where(context: 'topics').map { |tagging| tagging.tag.name }.uniq
```

Comment:
@iAmNathanJ  we can reference these variables in the views "@topics, @days .." they should provide up with the list of unique tags for each category